core/src/data/value/literal.rs

```diff
@@ -18,6 +18,22 @@ use {
     },
 };
 
+impl TryFrom<Literal<'_>> for Expr {
+    type Error = Error;
+
+    fn try_from(value: Literal<'_>) -> Result<Self> {
+        Expr::try_from(Value::try_from(value)?)
+    }
+}
+
+impl TryFrom<&Literal<'_>> for Expr {
+    type Error = Error;
+
+    fn try_from(value: &Literal<'_>) -> Result<Self> {
+        Expr::try_from(Value::try_from(value)?)
+    }
+}
```

Comment:
These would be also the ones that we don't need to use

Reply:
```rust
pub enum LiteralError {
    #[error("unsupported literal binary operation {} {} {}", .left, .op.to_sql(), .right)]
    UnsupportedBinaryOperation {
        left: String,
        op: BinaryOperator,
        right: String,
    },
// ...
```
```rust
pub enum EvaluateError {
// ...
    #[error("unsupported evaluate binary operation {} {} {}", .left, .op.to_sql(), .right)]
    UnsupportedBinaryOperation {
        left: String,
        op: BinaryOperator,
        right: String,
    },
// ...
```
I changed errors like above